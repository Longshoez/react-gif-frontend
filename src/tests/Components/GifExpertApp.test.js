import React from 'react';
import {shallow} from 'enzyme';
import {GifExpertApp} from '../../Components/GifExpertApp/GifExpertApp';

describe('Testing GifExpertApp Component', () => {

    let wrapper = shallow(<GifExpertApp />)
    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />)
    })

    test('component should render', () => {
        expect(wrapper).toMatchSnapshot()
    });


    test('should  show a category list', () => {
        //Arrange
        const categories = ['mandalorian', 'star wars']
        wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)        

        //Act

        //Assert
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    });
});