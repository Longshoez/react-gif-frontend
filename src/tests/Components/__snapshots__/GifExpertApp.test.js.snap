// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp Component component should render 1`] = `
<div>
  <div>
    <h2
      className="header"
    >
      Gif app ðŸ”¥
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <GifGrid
    category="Mandalorian"
    key="Mandalorian"
  />
</div>
`;

exports[`Testing GifExpertApp Component should  show a category list 1`] = `
<div>
  <div>
    <h2
      className="header"
    >
      Gif app ðŸ”¥
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <GifGrid
    category="mandalorian"
    key="mandalorian"
  />
  <GifGrid
    category="star wars"
    key="star wars"
  />
</div>
`;
